### ISSUE TITLE
is-half-mobile columns are not side by side

### ISSUE THREAD

**Issue by lukepighetti** [jgthms/bulma#1807]
`.column .is-half-mobile` columns are not displaying side-by-side. All other breakpoints work as anticipated.

![screen shot 2018-04-18 at 4 01 01 pm](https://user-images.githubusercontent.com/666539/38955139-b5b63d44-4321-11e8-9aa9-514a3229a68a.png)

```
<section class="section">
    <h1 class="title">Core</h1>
    <h2 class="subtitle">Lorem ipsum dolor sit amet, consectetur adipisicing.</h2>

    <div class="columns is-multiline">

      <div class="column is-half-mobile is-one-third-tablet is-one-quarter-desktop is-one-fifth-widescreen">
        <div class="card">
          <div class="card-image">
            <figure class="image is-square">
              <img src="img/luke-pighetti.jpg" alt="">
            </figure>
          </div>
          <div class="card-content">
            <div class="media">
              <div class="media-content">
                <p class="title is-4">John Smith</p>
                <p class="subtitle is-6">Founder</p>
              </div>
            </div>

            <div class="content">
              Lorem ipsum dolor sit amet, consectetur adipiscing elit.
              Phasellus nec iaculis mauris.
            </div>
          </div>

          <footer class="card-footer">
            <p class="card-footer-item">
              <span>
                <a href=""><i class="fab fa-linkedin fa-2x"></i></a>
              </span>
            </p>
            <p class="card-footer-item">
              <span>
                <a href=""><i class="fab fa-twitter-square fa-2x"></i></a>
              </span>
            </p>
          </footer>
        </div>
      </div>

      <div class="column is-half-mobile is-one-third-tablet is-one-quarter-desktop is-one-fifth-widescreen">
        <div class="card">
          <div class="card-image">
            <figure class="image is-square">
              <img src="img/gabriel-muresen.jpg" alt="">
            </figure>
          </div>
          <div class="card-content">
            <div class="media">
              <div class="media-content">
                <p class="title is-4">John Smith</p>
                <p class="subtitle is-6">Founder</p>
              </div>
            </div>

            <div class="content">
              Lorem ipsum dolor sit amet, consectetur adipiscing elit.
              Phasellus nec iaculis mauris.
            </div>
          </div>

          <footer class="card-footer">
            <p class="card-footer-item">
              <span>
                <a href=""><i class="fab fa-linkedin fa-2x"></i></a>
              </span>
            </p>
            <p class="card-footer-item">
              <span>
                <a href=""><i class="fab fa-twitter-square fa-2x"></i></a>
              </span>
            </p>
          </footer>
        </div>
      </div>

      <div class="column is-half-mobile is-one-third-tablet is-one-quarter-desktop is-one-fifth-widescreen">
              <div class="card">
                <div class="card-image">
                  <figure class="image is-square">
                    <img src="img/erik-mourin.jpg" alt="">
                  </figure>
                </div>
                <div class="card-content">
                  <div class="media">
                    <div class="media-content">
                      <p class="title is-4">John Smith</p>
                      <p class="subtitle is-6">Founder</p>
                    </div>
                  </div>

                  <div class="content">
                    Lorem ipsum dolor sit amet, consectetur adipiscing elit.
                    Phasellus nec iaculis mauris.
                  </div>
                </div>

                <footer class="card-footer">
                  <p class="card-footer-item">
                    <span>
                      <a href=""><i class="fab fa-linkedin fa-2x"></i></a>
                    </span>
                  </p>
                  <p class="card-footer-item">
                    <span>
                      <a href=""><i class="fab fa-twitter-square fa-2x"></i></a>
                    </span>
                  </p>
                </footer>
              </div>
            </div>


    </div>
  </section>
```

**Comment by nadim1992** [comment]
What about adding an extra is-mobile class on the columns div :)

<div class="columns is-multiline is-mobile">

**Comment by ivandov** [comment]
Though adding the `is-mobile` class to the columns div worked, this is not necessary to make the responsive classes work as expected for tablet or desktop viewports. 

I believe this inconsistency should be addressed.

**Comment by williamabbott** [comment]
Is this really the case? It's not very well thought out if so. There are responsive column helpers supposedly so that we can have multiple layouts depending on viewport size, but if we have to specify the .is-mobile on the columns wrapper, won't this fix the child columns to the mobile setting?

**Comment by yoyowinkler** [comment]
Following this advice ( adding `is-mobile`) on a 4 column layout renders the last 2 columns not visible.

```
<section class="container product-row">
      <div class="columns is-mobile">
        <div class="column is-one-quarter-desktop is-half-mobile is-half-tablet">
           <img src="images/photo1.jpg">
        </div>

        <div class="column is-one-quarter-desktop is-half-mobile is-half-tablet">
          <img src="images/photo2.jpg">
        </div>

        <div class="column is-one-quarter-desktop is-half-mobile is-half-tablet">
          <img src="images/photo3.jpg">
        </div>

        <div class="column is-one-quarter-desktop is-half-mobile is-half-tablet">
          <img src="images/photo4.jpg">
        </div>
      </div>
    </section>
```

**Comment by dredav** [comment]
@yoyowinkler You have to add the class `is-multiline` to the columns div. See more at https://bulma.io/documentation/columns/options/#multiline

**Comment by Hentaro1989** [comment]
Same here. I don't want to add `is-mobile`, since other classes like `is-one-fifth-desktop` will be not able to work. Does anyone have any ideas?

**Comment by Hentaro1989** [comment]
I solved it myself. And Sorry, I was using `is-half`. So it was different issue.

`is-one-fifth-desktop`  works. But `is-half` doesn't work on mobile size. The 6 div will drop out.
```html
<div class="columns is-multiline">
  <div class="column is-half is-one-fifth-desktop">1</div>
  <div class="column is-half is-one-fifth-desktop">2</div>
  <div class="column is-half is-one-fifth-desktop">3</div>
  <div class="column is-half is-one-fifth-desktop">4</div>
  <div class="column is-half is-one-fifth-desktop">5</div>
  <div class="column is-half is-one-fifth-desktop">6</div>
</div>
```

Add `is-mobile`, then `is-half` will work on mobile size. But `is-one-fifth-desktop` won't work...
```html
<div class="columns is-multiline is-mobile">
  ...
```

Change `is-half` to `is-half-touch`. It completely work!
```html
<div class="columns is-multiline is-mobile">
  <div class="column is-half is-one-fifth-desktop">1</div>
  <div class="column is-half is-one-fifth-desktop">2</div>
  <div class="column is-half is-one-fifth-desktop">3</div>
  <div class="column is-half is-one-fifth-desktop">4</div>
  <div class="column is-half is-one-fifth-desktop">5</div>
  <div class="column is-half is-one-fifth-desktop">6</div>
</div>
```

**Comment by jontroncoso** [comment]
`.is-mobile` should be added whenever you want to make the columns take affect in mobile. Expecting `.is-x-mobile` to work without a `.columns.is-mobile` parent is confusing and inconsistent with how the rest of bulma works. This ticket should be closed.

**Comment by pgmoir** [comment]
While the past responses vaguely answered my question, and I don't fully understand how it works, I wanted to accept that adding `is-mobile` works, but also explain for the benefit of others how to have different responsive column widths across everything from phone to ultra wide desktop, and alleviate the fears that using is-mobile will break the wider screens. It sort of does on its own, but I found that this solution works (and suspect that this is the intention, just doesn't seem to come across in the docs).

Mine is the classic card scenario, and the requirements are to have 2 cards across on smaller mobile devices, 3 cards wide on tablets portrait mode, 4 cards on landscape mode and desktop, 6 cards wide on full hd.

On my primary page, where the `columns` container lives, I had to set all the variants I would use. And then used the column size variable along with the device modifier. This seems to work as expected. I've consolidated my code into a single example. The card part in the middle would be your repeating code. I've only included one instance for brevity.

```
<div class="columns is-multiline is-mobile is-tablet is-desktop is-fullhd">
  <!-- cards (repeating) -->
  <div class="column is-6-mobile is-4-tablet is-3-desktop is-2-fullhd">
    <article class="card">
      <div class="card-image">
        <figure class="image is-4by3">
          <img src="" alt="" role="presentation" />
        </figure>
      </div>
      <div class="card-content">
        <p><strong>Blah</strong></p>
      </div>
    </article>
  </div>
  <!-- end of repeating card --> 
</div>
```
